### PR TITLE
CORE-6226 - publish records transactionally

### PR DESCRIPTION
**Pull request by nikinagy**
https://r3-cev.atlassian.net/browse/CORE-6226

### CODE REVIEW

File: components/membership/registration-impl/src/main/kotlin/net/corda/membership/impl/registration/staticnetwork/StaticMemberRegistrationService.kt [hunk 1]
```diff
@@ -134,8 +134,7 @@ class StaticMemberRegistrationService @Activate constructor(
             val groupPolicy = groupPolicyProvider.getGroupPolicy(member)
                 ?: throw CordaRuntimeException("Could not find group policy for member: [$member]")
             val (memberInfo, records) = parseMemberTemplate(member, groupPolicy, keyScheme)
-            records.publish()
-            listOf(createHostedIdentity(member, groupPolicy)).publish()
+            (records + createHostedIdentity(member, groupPolicy)).publish()
```
Comment (yift-r3) [inline components/membership/registration-impl/src/main/kotlin/net/corda/membership/impl/registration/staticnetwork/StaticMemberRegistrationService.kt:137]: Should we wait for the future to complete?

Comment (nikinagy) [reply components/membership/registration-impl/src/main/kotlin/net/corda/membership/impl/registration/staticnetwork/StaticMemberRegistrationService.kt:137]: We wait for it in the `publish()` function. See line 149.

**Review by yift-r3 — APPROVED** [review]
LGTM

File: components/membership/registration-impl/src/main/kotlin/net/corda/membership/impl/registration/staticnetwork/StaticMemberRegistrationService.kt (hunk 1 above)
Comment (yift-r3) [reply components/membership/registration-impl/src/main/kotlin/net/corda/membership/impl/registration/staticnetwork/StaticMemberRegistrationService.kt:137]: Ooops

**Pull request merged**